### PR TITLE
Add project joining modal

### PR DESCRIPTION
**Pull request by JoshSmith**
# What's in this PR?

This adds the project joining modal dialog.

The implementation is basically fully working. I might add more polish but this is essentially shippable in its current state.

Before merging, though, we should (at least):

- [x] Add page objects
- [x] Write component tests
- [x] Write acceptance tests
- [x] Clean up the template logic of the project card's buttons
- [x] Optionally remove the usage of `store` in the component
- [x] Remove tests for tooltips in `skill-list-item-link`
- [x] Maybe remove styling for these tooltips? If it exists

### CODE REVIEW

File: app/styles/components/project-join-modal.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+.project-join-modal__skills {
+  margin-bottom: 20px;
+  text-align: center;
+
+  p {
+    font-size: $body-font-size-normal;
+    margin-bottom: 10px;
+  }
+
+  .related-skills .skills {
```
Comment (JoshSmith) [inline app/styles/components/project-join-modal.scss]: I don't like that this styling is happening in here. Seems like there should be a more universal way to fix this.

File: app/templates/components/project-card.hbs [hunk 2]
```diff
@@ -13,6 +13,29 @@
...
+          {{/if}}
+        {{/if}}
+
+        {{#unless currentProjectMembership}}
+          <button class="button default small" {{action "toggleJoinProjectModal"}}>Join project</button>
+        {{/unless}}
+      {{else}}
+        {{link-to "Join project" "signup" class="button default small"}}
+      {{/if}}
+    {{/if}}
```
Comment (JoshSmith) [inline app/templates/components/project-card.hbs]: All of this logic is way too complex.

File: app/templates/components/project-join-modal.hbs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          <img class="icon medium" src={{user.photoLargeUrl}} />
+        </div>
+        <div class="project-join-modal__icons__icon project-join-modal__icons__icon--join"></div>
+        <div class="project-join-modal__icons__avatar">
+          <img class="icon medium" src={{project.iconLargeUrl}} />
+        </div>
+      </div>
+      <div class="project-join-modal__skills">
+        <p>What can you help with?</p>
+        {{related-skills isClickable=true skills=skills}}
```
Comment (JoshSmith) [inline app/templates/components/project-join-modal.hbs]: This `isClickable` is not at all tested on a couple of the higher-level components.

File: app/templates/components/project-card.hbs (hunk 2 above)
Comment (begedin) [inline app/templates/components/project-card.hbs]: I've simplified this some by moving the modal and the button that opens it into the same component. My suggestion for further simplification would be to do some extraction, but probably in a separate PR.

The two base cases are:

- There is a membership
- There is no membership

For the first one, show a `membership-status` component with 3 states

- loading
- request sent (role eq pending)
- member (role eq something else)

For the second one, we just use the current component.

```hbs
{{#if session.isAuthenticated}}
  {{#if currentMembership}}
    {{membership-status membership=currentMembership}}
  {{else}}
    {{project-join-modal onJoin=(action onJoin project) project=project skills=projectSkills}}
  {{/if}}
{{else}}
  {{link-to "Join project" "signup" class="button default small"}}
{{/if}}
```

**Comment by JoshSmith** [comment]
@begedin had mentioned this morning that there page objects for working with modals. Surfacing that here.

Do let me know if you'll be working on this further, and whenever you're done with it, assign over to me so I'm tracking.

**Comment by begedin** [comment]
@JoshSmith Ready for review. The button logic in project card was simplified some, but there's more room for improvement. Probably better done as part of another PR.

Also, while doing this, I noticed the project-card component is severely under-tested.

Before merge:

- [ ] Create issue: Simplify project card button logic
- [ ] Create issue: Write integration tests for project card

File: app/templates/components/project-card.hbs (hunk 2 above)
Comment (JoshSmith) [inline app/templates/components/project-card.hbs]: What do you think about this being called `project-user-status` Instead?

**Comment by JoshSmith** [comment]
I also added the modal to the project page itself. We need to fix the duplication here.

@begedin I'm wondering if we should open an issue to move the actions defined as `joinProject` up into either a service, or even into the application controller. I think maybe the latter makes the most sense, actually?

**Comment by JoshSmith** [comment]
I refactored the `joinProject` action to instead call a `projectUser` service with `joinProject`.

**Comment by JoshSmith** [comment]
We need to create an issue to write project user service tests @begedin.

**Pull request merged**